src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -984,6 +985,10 @@ trait Contexts { self: Analyzer =>
 
     def isNameInScope(name: Name) = lookupSymbol(name, _ => true).isSuccess
 
+    def isPlainClassInJavaUnit(sym: Symbol): Boolean =
```

Comment:
How about:

```java
class HasStatic { static void foo() {} }
class Client { void test() { ScalaSubclass.foo(); } }

```

```scala
class ScalaSubclass extends HasStatic
```

This shows that a Java client can access a Java static selected from a Scala subclass of the owner of the static member.

I think we should drop the check for `sym.isJavaDefined` here, and instead only use this to filter out companions in `FindMember`.